hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryManager.java

```diff
@@ -73,6 +83,23 @@ public void restartTrackingFrom(FCMap<MerkleEntityId, MerkleAccount> accounts) {
 		txnHistories.values().forEach(TxnIdRecentHistory::observeStaged);
 	}
 
+	public void restartEntitiesTrackingFrom(FCMap<MerkleEntityId, MerkleSchedule> schedules) {
+		entityExpiries.reset();
+
+		var expiries = new ArrayList<Map.Entry<Pair<Long, Consumer<EntityId>>, Long>>();
+		schedules.forEach((id, schedule) -> {
+			if (!schedule.isDeleted()) {
+				var pair = new Pair<Long, Consumer<EntityId>>(id.getNum(), scheduleStore::expire);
+				expiries.add(new AbstractMap.SimpleImmutableEntry<>(pair, schedule.expiry()));
+			}
+		});
+		// todo: add accounts, files, tokens, topics
+
+		var cmp = Comparator.comparing(Map.Entry<Pair<Long, Consumer<EntityId>>, Long>::getValue);
```

Comment:
Since it's possible for multiple entities to expire at the same time, we need this sort to break ties in a deterministic way. (Otherwise at a given second, all nodes might remove the same entities from the `FCMap`, but _in a different order_; causing an ISS). So e.g.

```
var cmp = Comparator.comparing(Map.Entry<Pair<Long, Consumer<EntityId>>, Long>::getValue)
    .thenComparing(entry -> entry.getKey().getLeft());
```

This guarantees that all nodes will remove entities expiring in the same second in order of increasing entity id num.